gms/gossiper.cc

```diff
@@ -2154,24 +2158,18 @@ future<> gossiper::add_saved_endpoint(inet_address ep, permit_id pid) {
         logger.debug("not replacing a previous ep_state for {}, but reusing it: {}", ep, ep_state);
         ep_state.update_timestamp();
     }
-    // It's okay to use the local version generator for the loaded application state values
-    // As long as the endpoint_state has zero generation.
-    // It will get updated as a whole by handle_major_state_change
-    // via do_apply_state_locally when (remote_generation > local_generation)
-    const auto tmptr = get_token_metadata_ptr();
-    auto host_id = tmptr->get_host_id_if_known(ep);
-    if (host_id) {
-        ep_state.add_application_state(gms::application_state::HOST_ID, versioned_value::host_id(host_id.value()));
-        auto tokens = tmptr->get_tokens(*host_id);
-        if (!tokens.empty()) {
-            std::unordered_set<dht::token> tokens_set(tokens.begin(), tokens.end());
-            ep_state.add_application_state(gms::application_state::TOKENS, versioned_value::tokens(tokens_set));
-        }
+    ep_state.add_application_state(gms::application_state::HOST_ID, versioned_value::host_id(loaded_state.host_id));
+    if (!loaded_state.tokens.empty()) {
+        ep_state.add_application_state(gms::application_state::TOKENS, versioned_value::tokens(loaded_state.tokens));
+    }
```

Comment:
Is making these parts unconditional related to the rack/dc? If so please explain in the changelog.

Reply:
okay, I'll add an explanation.
Looking up the host_id is up to the caller now.
And basically the tokens, dc, and rack are useless without host_id.

Reply:
The host_id can be made mandatory after 50ebce8acc2406bcc8466b2f051e2ceae395e83e